src_firmware/src/ble_gatt.c

```diff
@@ -5,17 +5,79 @@
 // TODO - move define wrappers to CMakeLists. If whole is not to be compiled, make rule in CMAKE
 
 #include "driver.h"
+#include "storage.h"
+#include <stdlib.h>
 
 #include <zephyr/kernel.h>
 #include <zephyr/bluetooth/gatt.h>
+#include "return_codes.h"
 
-static char ble_buff[8];
+#define UUID_GENERATOR(val)	BT_UUID_DECLARE_128(BT_UUID_128_ENCODE(val,		\
+								       0x37de,		\
+								       0x44d4,		\
+								       0xbe45,		\
+								       0xd1f1fd9385fd))
 
-static ssize_t read_ble(struct bt_conn *conn, const struct bt_gatt_attr *attr,
+
+#define SERVICE_UUID			BT_UUID_DECLARE_16(0xffaa)
+#define TX_CHAR_UUID_COMMAND		UUID_GENERATOR(0x6e006610)
+#define RX_CHAR_UUID_TEMPL_LIST		UUID_GENERATOR(0x6e006620)
+#define RX_CHAR_UUID_TEMPL_ACTIVE	UUID_GENERATOR(0x6e006621)
+#define RX_CHAR_UUID_SPEED		UUID_GENERATOR(0x6e006622)
+#define RX_CHAR_UUID_HW_VER		UUID_GENERATOR(0x6e006623)
+
+
+
+
+static ssize_t read_ble_template_list(struct bt_conn *conn, const struct bt_gatt_attr *attr,
+			 void *buf, uint16_t len, uint16_t offset)
+{
+	ssize_t ret;
+	char *buffer;
+	uint8_t template_count = get_template_size();
+	size_t frame_size = template_count * (sizeof(uint32_t) + CONFIG_TEMPLATE_NAME_SIZE) + 4;
+
+	buffer = (char *)calloc(frame_size, sizeof(char));
+
+	buffer[0] = frame_size;
+	buffer[1] = 0; //page number
```

Comment:
please add TODO comment- we should implement paging for more than 15 templates in the future